dspace-server-webapp/src/test/java/org/dspace/app/rest/RequestItemRepositoryIT.java

```diff
@@ -0,0 +1,368 @@
+
+        // Create it and see if it was created correctly.
+        ObjectMapper mapper = new ObjectMapper();
+        MvcResult mvcResult = getClient()
+                .perform(post(URI_ROOT)
+                        .content(mapper.writeValueAsBytes(rir))
+                        .contentType(contentType))
+                .andExpect(status().isCreated())
+                .andExpect(content().contentType(contentType))
+                .andExpect(jsonPath("$", Matchers.allOf(
+                        hasJsonPath("$.id", not(is(emptyOrNullString()))),
+                        hasJsonPath("$.type", is(RequestItemRest.NAME)),
+                        hasJsonPath("$.token", not(is(emptyOrNullString()))),
+                        hasJsonPath("$.requestEmail", is(RequestItemBuilder.REQ_EMAIL)),
+                        hasJsonPath("$.requestMessage", is(RequestItemBuilder.REQ_MESSAGE)),
+                        hasJsonPath("$.requestName", is(RequestItemBuilder.REQ_NAME)),
+                        hasJsonPath("$.allfiles", is(false)),
+                        hasJsonPath("$.requestDate", not(is(emptyOrNullString()))), // TODO should be an ISO datetime
+                        hasJsonPath("$._links.self.href", not(is(emptyOrNullString())))
+                )))
+                .andReturn();
+
+        // Clean up the created request.
+        String content = mvcResult.getResponse().getContentAsString();
+        Map<String,Object> map = mapper.readValue(content, Map.class);
+        String requestToken = String.valueOf(map.get("token"));
+        RequestItem ri = requestItemService.findByToken(context, requestToken);
+        requestItemService.delete(context, ri);
+
+        context.restoreAuthSystemState();
+    }
+
+    /**
+     * Test of getDomainClass method, of class RequestItemRepository.
+     */
+    @Test
+    public void testGetDomainClass() {
+        System.out.println("getDomainClass");
+        RequestItemRepository instance = new RequestItemRepository();
+        Class instanceClass = instance.getDomainClass();
+        assertEquals("Wrong domain class", RequestItemRest.class, instanceClass);
+    }
+
+    /**
+     * Verify that Spring Security's CSRF protection is working as we expect.
+     * We must test this using a simple non-GET request, as CSRF Tokens are not
+     * validated in a GET request.
+     *
+     * @throws java.lang.Exception passed through.
+     */
+    @Test
+    public void testRefreshTokenWithInvalidCSRF()
+            throws Exception {
+        // Login via password to retrieve a valid token
+        String token = getAuthToken(eperson.getEmail(), password);
+
+        // Remove "Bearer " from that token, so that we are left with the token itself
+        token = token.replace("Bearer ", "");
+
+        // Save token to an Authorization cookie
+        Cookie[] cookies = new Cookie[1];
+        cookies[0] = new Cookie(AUTHORIZATION_COOKIE, token);
+
+        getClient().perform(post(URI_ROOT)
```

Comment:
This POST appears to be posting nothing... what does it normally return when you post nothing? We should perhaps have a separate test with an invalid POST to show that it does NOT throw a "Forbidden" error (I doubt it would, but having a test that shows how an invalid POST would respond would be nice anyways)

Reply:
Yes, that should be posting a fresh item request.